source/_posts/2016-04-25-opencomponents-microservices-in-the-front-end-world.markdown

```diff
@@ -0,0 +1,259 @@
+---
+layout: post
+title: "OpenComponents - microservices in the front-end world"
+date: 2016-04-25 10:10:03 +0000
+comments: true
+author: mfigus
+categories: [SOA, Microsites, Microservices, OpenComponents, OC] 
+---
+
+Many engineers work every day on Opentable.com from our offices located in Europe, America, and Asia, pushing changes to production multiple times a day. Usually, this is very hard to achieve, in fact it took years for us to get to this point. I explained in a [previous article][1] how we dismantled our monolith in favour of a Microsites architecture. In the meanwhile, during the last two years we have been working on something I would define quite unique, given its purpose and its definition, called **OpenComponents**.
+
+### Another front-end framework?
+
+OpenComponents is a system for allowing people to have tools to facilitate code sharing, reduce dependencies, and easily approach new features and experiments from the back-end to the front-end. To achieve this, it is based on the concept of using services as interfaces for enabling pages to render partial contents located, executed and deployed independently.
+
+OpenComponents is not *another SPA JS framework*: it is a set of conventions, patterns and tools to develop and quickly deploy fragments of front-end. In this perspective, it plays nicely with any existing architecture and framework in terms of front-end and back-end: its purpose is to **serve as delivery mechanism for a more modularised end-result in the front-end**.
+
+OC is been in production for more than a year at OpenTable and it is [fully open-sourced][2].
+
+## Overview
+
+OpenComponents involves two parts:
+
+* The **consumers** are web pages that need fragments of html for rendering partial contents. Sometimes they need some content during server-side rendering, somethings when executing code in the browser.
+* The **components** are small units of isomorphic code mainly consisting of html, javascript, css. They can optionally contain some logic, allowing a server-side node.js closure to compose a model that is used to render the view. When rendered they are pieces of pure html, ready to be injected in any web page.
```

Comment:
i wonder what do you mean by 'pure' html? you could include javascript in this html if you wanted to, right?